internal/sveltos/profile.go

```diff
@@ -50,96 +51,149 @@ type HelmChartOpts struct {
+func Spec(opts *ReconcileProfileOpts) (*sveltosv1beta1.Spec, error) {
 	tier, err := PriorityToTier(opts.Priority)
 	if err != nil {
 		return nil, err
 	}
 
-	operation, err := ctrl.CreateOrUpdate(ctx, cl, cp, func() error {
-		if cp.Labels == nil {
-			cp.Labels = make(map[string]string)
-		}
+	spec := &sveltosv1beta1.Spec{
+		ClusterSelector: libsveltosv1beta1.Selector{
+			LabelSelector: opts.LabelSelector,
+		},
+		Tier:               tier,
+		ContinueOnConflict: !opts.StopOnConflict,
+	}
 
-		cp.Labels[hmc.HMCManagedLabelKey] = hmc.HMCManagedLabelValue
-		if opts.OwnerReference != nil {
-			cp.OwnerReferences = []metav1.OwnerReference{*opts.OwnerReference}
+	for _, hc := range opts.HelmChartOpts {
+		helmChart := sveltosv1beta1.HelmChart{
+			RepositoryURL:    hc.RepositoryURL,
+			RepositoryName:   hc.RepositoryName,
+			ChartName:        hc.ChartName,
+			ChartVersion:     hc.ChartVersion,
+			ReleaseName:      hc.ReleaseName,
+			ReleaseNamespace: hc.ReleaseNamespace,
+			HelmChartAction:  sveltosv1beta1.HelmChartActionInstall,
+			RegistryCredentialsConfig: &sveltosv1beta1.RegistryCredentialsConfig{
+				PlainHTTP:             hc.PlainHTTP,
+				InsecureSkipTLSVerify: hc.InsecureSkipTLSVerify,
+			},
 		}
 
-		cp.Spec = sveltosv1beta1.Spec{
-			ClusterSelector: libsveltosv1beta1.Selector{
-				LabelSelector: metav1.LabelSelector{
-					MatchLabels: matchLabels,
-				},
-			},
-			Tier:               tier,
-			ContinueOnConflict: !opts.StopOnConflict,
+		if hc.PlainHTTP {
+			// InsecureSkipTLSVerify is redundant in this case.
+			helmChart.RegistryCredentialsConfig.InsecureSkipTLSVerify = false
 		}
 
-		for _, hc := range opts.HelmChartOpts {
-			helmChart := sveltosv1beta1.HelmChart{
-				RepositoryURL:    hc.RepositoryURL,
-				RepositoryName:   hc.RepositoryName,
-				ChartName:        hc.ChartName,
-				ChartVersion:     hc.ChartVersion,
-				ReleaseName:      hc.ReleaseName,
-				ReleaseNamespace: hc.ReleaseNamespace,
-				HelmChartAction:  sveltosv1beta1.HelmChartActionInstall,
-				RegistryCredentialsConfig: &sveltosv1beta1.RegistryCredentialsConfig{
-					PlainHTTP:             hc.PlainHTTP,
-					InsecureSkipTLSVerify: hc.InsecureSkipTLSVerify,
-				},
+		if hc.Values != nil {
+			b, err := hc.Values.MarshalJSON()
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal values to JSON for service %s: %w", hc.RepositoryName, err)
 			}
 
-			if hc.PlainHTTP {
-				// InsecureSkipTLSVerify is redundant in this case.
-				helmChart.RegistryCredentialsConfig.InsecureSkipTLSVerify = false
+			b, err = yaml.JSONToYAML(b)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert values from JSON to YAML for service %s: %w", hc.RepositoryName, err)
 			}
 
-			if hc.Values != nil {
-				b, err := hc.Values.MarshalJSON()
-				if err != nil {
-					return fmt.Errorf("failed to marshal values to JSON for service (%s) in ManagedCluster: %w", hc.RepositoryName, err)
-				}
+			helmChart.Values = string(b)
```

Comment:
performance nit: if we expect to have long strings then maybe converting to string without allocation might be helpful here (i.e. `unsafe.String(&b[0], b)`)